tensorflow_quantum/core/src/circuit_parser.cc

```diff
@@ -159,4 +159,45 @@ Status CircuitFromPauliTerm(const tfq::proto::PauliTerm& term,
   return CircuitFromProgram(measurement_program, num_qubits, circuit);
 }
 
+Status ZBasisCircuitFromPauliTerm(const tfq::proto::PauliTerm& term,
+                                  const int num_qubits, Circuit* circuit) {
+  Program measurement_program;
+  measurement_program.mutable_circuit()->set_scheduling_strategy(
+      cirq::google::api::v2::Circuit::MOMENT_BY_MOMENT);
+  Moment* term_moment = measurement_program.mutable_circuit()->add_moments();
+  float transform_exponent = 0.0;
+  std::string gate_type;
+  for (const tfq::proto::PauliQubitPair& pair : term.paulis()) {
+    if (pair.pauli_type() == "Z") {
+      // Z requires no transform.
+      continue;
+    }
+
+    // Assume it is X and the transform is Y^-0.5
+    transform_exponent = -0.5;
+    gate_type = "Y";
+    if (pair.pauli_type() == "Y") {
+      // Y regquires X**0.5 transform.
+      transform_exponent = 0.5;
+      gate_type = "X";
+    }
+
+    Operation* new_op = term_moment->add_operations();
+
+    // create corresponding eigen gate op.
+    new_op->add_qubits()->set_id(pair.qubit_id());
+    new_op->mutable_gate()->set_id(gate_type + "P");
+    (*new_op->mutable_args())["exponent"].mutable_arg_value()->set_float_value(
+        transform_exponent);
+    (*new_op->mutable_args())["global_shift"]
+        .mutable_arg_value()
+        ->set_float_value(0.0);
+    (*new_op->mutable_args())["exponent_scalar"]
+        .mutable_arg_value()
+        ->set_float_value(1.0);
+  }
+
+  return CircuitFromProgram(measurement_program, num_qubits, circuit);
```

Comment:
When we used these circuits in V0.1, we also needed their inverses.  Is there an easy way to take circuit inverses with our protos?

Reply:
I don't know if I follow here exactly. Why do we need inverses here ?

Reply:
It depends on what this function is supposed to be used for.  In V0.1, our `many_clifford_to_many_z` function returned the circuit and its inverse, so that we could place coefficients as Z rotations at the base of CNOT ladders.  I suppose here we will only be using this function to transform the state to the measurement basis?

Reply:
Yes, the only use case for this I am imagining is transforming the basis on the scratch wavefunction for each term in a bigger PauliSum so to estimate the expecation you might do:

1. copy ref onto scratch.
2. evolve scratch forward with circuit from this.
3. sample and estimate expectation.
4. go to step 1 with next term in the PauliSum.